File: src/main/webapp/resources/app/SearchService.js
mainModule.service('SearchService', ['$http', function ($http) {
    return {
        searchBooksByTitle: function (title) {
            return $http.get("searchBooksByTitle/" + title)
                .then(function (response) {
                    return response.data;
                });
        },
        searchBooksByTag: function (tag) {
            return $http.get("searchBooksByTag/" + tag)
                .then(function (response) {
                    return response.data;
                });
        },
        searchAuthorsByName: function (name) {
            return $http.get("searchAuthorsByName/" + name)
                .then(function (response) {
                    return response.data;
                });
        }
    }
}]);

File: service/src/main/java/com/readit/service/SearchService.java
package com.readit.service;

import com.readit.entity.Author;
import com.readit.entity.Book;

import java.util.List;

public interface SearchService {

    List<Book> findBooksByTitle(String title);

    List<Book> findBooksByTag(String tag);

    List<Author> findAuthorsByName(String name);
}
